### DISCUSSION TITLE
急需帮助  umi3 中的 qiankun 怎么用 我这个项目要作为子应用 需要更换dom 节点 怎么更换啊

### DISCUSSION THREAD

**Discussion by shxinlei** [umijs/umi#6428] [Q&A]
export const qiankun = {
  // 应用加载之前
  async bootstrap(props:any) {
    
    if(props){
      console.log('app1 bootstrap', props);
      // querySelector("singlespa-container")  // 更换为这个 dom 节点
    }
    // return props
    // debugger;
  },
  // 应用 render 之前触发
  async mount(props: any) {

    console.log('app1 mount', props);
    return props
  },
  // 应用卸载之后触发
  async unmount(props:any) {
    console.log('app1 unmount', props);
    return props
  },
};

**Comment by ChengranYan** [comment]
同求，能不能把render暴露出来呢
